hooks/tk-multi-publish2/basic/nuke_update_flame_clip.py

```diff
@@ -372,82 +433,96 @@ def _update_flame_clip(self, item):
 
         self.logger.info("Updating Flame clip file...")
 
-        # get a handle on the write node app, stored during accept()
-        write_node_app = item.properties["sg_writenode_app"]
-
-        # each publish task is connected to a nuke write node instance. this
-        # value was populated via the collector and verified during accept()
-        write_node = item.properties["sg_writenode"]
-
         # get the clip path as processed during validate()
         flame_clip_path = item.properties["flame_clip_path"]
 
-        # get the fields from the work file
-        render_path = write_node_app.get_node_render_path(write_node)
-        render_template = write_node_app.get_node_render_template(write_node)
-        render_path_fields = render_template.get_fields(render_path)
-        publish_template = write_node_app.get_node_publish_template(write_node)
-
-        # set up the sequence token to be Flame friendly
-        # e.g. mi001_scene_output_v001.[0100-0150].dpx
-        # note - we cannot take the frame ranges from the write node -
-        # because those values indicate the intended frame range rather
-        # than the rendered frame range! In order for Flame to pick up
-        # the media properly, it needs to contain the actual frame data
-
-        # get all paths for all frames and all eyes
-        paths = self.parent.sgtk.paths_from_template(
-            publish_template,
-            render_path_fields,
-            skip_keys=["SEQ", "eye"]
-        )
-
-        # for each of them, extract the frame number. Track the min and the max.
-        # TODO: would be nice to have a convenience method in core for this.
-        min_frame = None
-        max_frame = None
-        for path in paths:
-            fields = publish_template.get_fields(path)
-            frame_number = fields["SEQ"]
-            if min_frame is None or frame_number < min_frame:
-                min_frame = frame_number
-            if max_frame is None or frame_number > max_frame:
-                max_frame = frame_number
-
-        # ensure we have a min/max frame
-        if min_frame is None or max_frame is None:
-            raise Exception(
-                "Couldn't extract min and max frame from the published "
-                "sequence! Will not update Flame clip xml."
+        # get a handle on the write node app, stored during accept()
+        write_node_app = item.properties.get("sg_writenode_app")
+        render_path_fields = None
+
+        if not write_node_app:
+            # If we don't have a writenode, we just parse the first frame of the
+            # item's sequence path to get the start and end frames.
+            publish_path_flame = _get_flame_frame_spec_from_path(
```

Comment:
seems this should be in the path info utils class in publish? What do you think @josh-t ?

Reply:
If I'm reading correctly, a method that, given a path with a frame number, returns a path where the frame number is replaced with a spec. We have [this](http://developer.shotgunsoftware.com/tk-multi-publish2/utility.html#tk_multi_publish2.util.get_frame_sequence_path) already, but it's not quite what is needed here. I think we could certainly add something like this, but the more I look at `path_info`, the more I think we need to really consider long term how it fits with the strings/template system. My inclination is to let this live here for the time being.